subprojects/core/src/test/groovy/org/gradle/api/internal/project/DefaultProjectTest.groovy

```diff
@@ -132,141 +113,122 @@ class DefaultProjectTest {
 
     groovy.lang.Script testScript
 
-    ScriptSource script = context.mock(ScriptSource.class)
+    ScriptSource script = Stub(ScriptSource)
 
     ServiceRegistry serviceRegistryMock
     ServiceRegistryFactory projectServiceRegistryFactoryMock
-    TaskContainerInternal taskContainerMock = context.mock(TaskContainerInternal.class)
-    Factory<AntBuilder> antBuilderFactoryMock = context.mock(Factory.class)
+    TaskContainerInternal taskContainerMock = Stub(TaskContainerInternal)
+    Factory<AntBuilder> antBuilderFactoryMock = Stub(Factory)
     AntBuilder testAntBuilder
 
-    ConfigurationContainer configurationContainerMock = context.mock(ConfigurationContainer.class)
-    RepositoryHandler repositoryHandlerMock = context.mock(RepositoryHandler.class)
-    DependencyHandler dependencyHandlerMock = context.mock(DependencyHandler)
-    ComponentMetadataHandler moduleHandlerMock = context.mock(ComponentMetadataHandler)
-    ScriptHandler scriptHandlerMock = context.mock(ScriptHandler)
-    DependencyMetaDataProvider dependencyMetaDataProviderMock = context.mock(DependencyMetaDataProvider)
-    GradleInternal build = context.mock(GradleInternal)
-    ConfigurationTargetIdentifier configurationTargetIdentifier = context.mock(ConfigurationTargetIdentifier)
-    FileOperations fileOperationsMock = context.mock(FileOperations)
-    ProviderFactory propertyStateFactoryMock = context.mock(ProviderFactory)
-    ProcessOperations processOperationsMock = context.mock(ProcessOperations)
-    LoggingManagerInternal loggingManagerMock = context.mock(LoggingManagerInternal.class)
-    Instantiator instantiatorMock = context.mock(Instantiator)
-    SoftwareComponentContainer softwareComponentsMock = context.mock(SoftwareComponentContainer.class)
-    InputNormalizationHandler inputNormalizationHandler = context.mock(InputNormalizationHandler.class)
-    ProjectConfigurationActionContainer configureActions = context.mock(ProjectConfigurationActionContainer.class)
-    PluginManagerInternal pluginManager = context.mock(PluginManagerInternal.class)
-    PluginContainer pluginContainer = context.mock(PluginContainer.class)
-    ManagedProxyFactory managedProxyFactory = context.mock(ManagedProxyFactory.class)
-    AntLoggingAdapter antLoggingAdapter = context.mock(AntLoggingAdapter.class)
-    AttributesSchema attributesSchema = context.mock(AttributesSchema)
-    TextResourceLoader textResourceLoader = context.mock(TextResourceLoader)
+    ConfigurationContainer configurationContainerMock = Stub(ConfigurationContainer)
```

Comment:
Mockmockmockmockmockmockmockmock....

Reply:
It's mocking half of Gradle to verify conventions and basic behavior of `DefaultProject`. And yes, it's not exactly pretty.